src/role.ts

```diff
@@ -213,282 +193,299 @@ export class Role extends RoleBase implements IRole {
     binding.addSubjects(...subjects);
     return binding;
   }
+
+  private synthesizeRules(): k8s.PolicyRule[] {
+    const rules: k8s.PolicyRule[] = [];
+    for (const rule of this._rules) {
+      for (const resource of rule.resources) {
+        rules.push({
+          verbs: rule.verbs,
+          apiGroups: [resource.apiGroup === 'core' ? '' : resource.apiGroup],
+          resourceNames: resource.resourceName ? [resource.resourceName] : undefined,
+          resources: resource.resourceType ? [resource.resourceType] : undefined,
```

Comment:
nit: I think we might be able to simplify this
```suggestion
          resources: resource.resourceType,
```